invenio_vocabularies/datastreams/readers.py

```diff
@@ -224,19 +224,31 @@ def _iter(self, fp, *args, **kwargs):
 class XMLReader(BaseReader):
     """XML reader."""
 
+    def __init__(self, root_element=None, *args, **kwargs):
```

Comment:
Make it mandatory like:

```suggestion
    def __init__(self, root_element, *args, **kwargs):
```

This is a breaking change and needs a major bump (I think it was already)

Reply:
If I remember well, making it mandatory failed due to the way that the factories work.
This is why all readers/transformers/writers have their arguments defaulting to `None`.